azext_iot/digitaltwins/providers/connection/builders.py

```diff
@@ -44,20 +53,37 @@ def __init__(
         eh_resource_group: str,
         eh_subscription: str,
         eh_consumer_group: str = DEFAULT_CONSUMER_GROUP,
+        identity: str = SYSTEM_IDENTITY,
         yes: bool = False,
     ):
         self.cli = EmbeddedCLI()
         self.yes = yes
         self.dt = dt_instance
-        if self.dt.identity is None:
-            raise BadRequestError(DT_IDENTITY_ERROR)
+        # Check that the identity is associated with the dt
+        if self.dt.identity is None and identity == SYSTEM_IDENTITY:
+            raise BadRequestError(DT_SYS_IDENTITY_ERROR)
+        elif (
+            identity != SYSTEM_IDENTITY and (
+                self.dt.identity is None
+                or self.dt.identity.user_assigned_identities is None
+                or identity not in self.dt.identity.user_assigned_identities
+            )
+        ):
+            raise BadRequestError(DT_UAI_IDENTITY_ERROR)
```

Comment:
nit - I think we can simply the logic here a bit with a single nested `if`:
Another nit/question - do we need to check `is None` or is `not self.dt.identity` sufficient?

```suggestion
        # Check that the identity is associated with the dt
        if not self.dt.identity:
            if identity == SYSTEM_IDENTITY:
                raise BadRequestError(DT_SYS_IDENTITY_ERROR)
            elif (
                not self.dt.identity.user_assigned_identities
                or identity not in self.dt.identity.user_assigned_identities
            ):
                raise BadRequestError(DT_UAI_IDENTITY_ERROR)
```

Reply:
Your if will not work... but I can change the `is None` to just having a `not`

This logic takes me a bit to process every time...

1. if there is no dt.identity, then you cannot do anything
  a. if you are trying for system, show the system error
  b. if you are trying for user, show the user error
2. if there is an identity, and you are trying for user, you need to check if there are any user identities associated with the dt and the user identity asked for is there too
  a. if there is no user identities or the user identity asked for isnt there, show the user error

The logic above will result in having `raise BadRequestError(DT_UAI_IDENTITY_ERROR)` in two places,  so I reworked that so it would be there in one place. If you think this logic is more reasonable, I can change the code back to it (we will have to deal with the same error raised in two places next to each other)

Reply:
You're correct, I missed one step of the `self.dt.identity` check that would end up resulting in managing two separate exception cases 👍